### PR TITLE
[Keyboard] Valor Rev3

### PR DESCRIPTION
**Pull request by Xelus22**
## Description

Add Valor Rev3

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Add Valor Rev3

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/xelus/valor/rev3/config.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/* key matrix size */
+#define MATRIX_ROWS 5
+#define MATRIX_COLS 15
```
Comment (drashna) [inline keyboards/xelus/valor/rev3/config.h:21]: Shouldn't be needed.
```suggestion
```

File: keyboards/xelus/valor/rev3/info.json [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgblight": true
+    },
+    "rgblight": {
+        "led_count": 32,
+        "pin": "B0",
```
Comment (drashna) [inline keyboards/xelus/valor/rev3/info.json:29]: This will need to be migrated to ws2812.pin on develop (no action needed, currently)

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/xelus/valor/rev3/config.h [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+#pragma once
+
+/* RGB Underglow */
+#define RGBLIGHT_DEFAULT_MODE (RGBLIGHT_MODE_RAINBOW_SWIRL + 2)
+#define RGBLIGHT_LIMIT_VAL 200
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
```
Comment (fauxpark) [inline keyboards/xelus/valor/rev3/config.h:25]:
```suggestion
```

File: keyboards/xelus/valor/rev3/rev3.h [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+#define ____ KC_NO
+
+#define LAYOUT_alice_split_bs( \
```
Comment (fauxpark) [inline keyboards/xelus/valor/rev3/rev3.h:23]: Should be moved to info.json.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/xelus/valor/rev3/rev3.h [hunk 5]
```diff
@@ -0,0 +1,20 @@
+/* Copyright 2023 Harrison Chan (Xelus)
```
Comment (fauxpark) [inline keyboards/xelus/valor/rev3/rev3.h:1]: This file can be deleted.

File: keyboards/xelus/valor/rev3/rev3.c [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// Nothing to see here, move along... ;-)
+
+#include "rev3.h"
```
Comment (fauxpark) [inline keyboards/xelus/valor/rev3/rev3.c:19]:
```suggestion
#include "quantum.h"
```

File: keyboards/xelus/valor/rev3/readme.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+    make xelus/valor/rev3:default
+
+Reset your keyboard in 3 ways:
+<ol>
+<li>Software reset on Fn + PgDn (bottom of the 3 macro buttons)</li>
+<li>Bootmagic reset: hold down the top left key (usually escape) and plugin the keyboard</li>
+<li>Physical reset button: on the back of the PCB, there should be a small golden button you can press</li>
+</ol>
+
```
Comment (fauxpark) [inline keyboards/xelus/valor/rev3/readme.md:19]:
```suggestion
```

File: keyboards/xelus/valor/rev3/readme.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (fauxpark) [inline keyboards/xelus/valor/rev3/readme.md:20]:
```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
```

File: keyboards/xelus/valor/rev3/info.json [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+            "christmas": true,
+            "knight": true,
+            "rainbow_mood": true,
+            "rainbow_swirl": true,
+            "rgb_test": true,
+            "snake": true,
+            "static_gradient": true,
+            "twinkle": true
+        }
+    },
```
Comment (fauxpark) [inline keyboards/xelus/valor/rev3/info.json:45]:
```suggestion
    },
    "community_layouts": ["alice_split_bs"],
```

File: keyboards/xelus/valor/rev3/rules.mk [hunk 10]
```diff
@@ -0,0 +1,3 @@
+LTO_ENABLE = yes
+
+LAYOUTS = alice_split_bs
```
Comment (fauxpark) [inline keyboards/xelus/valor/rev3/rules.mk:3]:
```suggestion
```

File: keyboards/xelus/valor/rev3/info.json [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgblight": true
+    },
```
Comment (fauxpark) [inline keyboards/xelus/valor/rev3/info.json:26]:
```suggestion
    },
    "build": {
        "lto": true
    },
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/xelus/valor/rev3/config.h [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/* RGB Underglow */
+#define RGBLIGHT_DEFAULT_MODE (RGBLIGHT_MODE_RAINBOW_SWIRL + 2)
+#define RGBLIGHT_LIMIT_VAL 200
```
Comment (dunk2k) [inline keyboards/xelus/valor/rev3/config.h:21]: migrate to `info.json`
```suggestion
```

File: keyboards/xelus/valor/rev3/info.json [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+    "build": {
+        "lto": true
+    },
+    "ws2812": {
+        "pin": "B0"
+    },
+    "rgblight": {
+        "led_count": 32,
+        "hue_steps": 8,
```
Comment (dunk2k) [inline keyboards/xelus/valor/rev3/info.json:35]: default setting (source:`/quantum/rgblight/rgblight.h`)
```suggestion
```

File: keyboards/xelus/valor/rev3/info.json [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+        "saturation_steps": 8,
+        "brightness_steps": 8,
```
Comment (dunk2k) [inline keyboards/xelus/valor/rev3/info.json:37]: migrated from `config.h`
```suggestion
        "brightness_steps": 8,
        "max_brightness": 200,
```

File: keyboards/xelus/valor/rev3/keymaps/default/readme.md
Comment (dunk2k) [inline keyboards/xelus/valor/rev3/keymaps/default/readme.md:1]: Current content of this file serves no informative purpose, could this file be removed?

File: keyboards/xelus/valor/rev3/keymaps/via/readme.md
Comment (dunk2k) [inline keyboards/xelus/valor/rev3/keymaps/via/readme.md:1]: Current content of this file serves no informative purpose, could this file be removed?

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by Xelus22** [comment]
weird i cant reopen this?

**Pull request closed without merging**